ConfigSpace/configuration_space.py

```diff
@@ -45,20 +45,32 @@
 
 
 class ConfigurationSpace(object):
+
     # TODO add comments to both the configuration space and single
     # hyperparameters!
 
     # TODO add a method to add whole configuration spaces as a child "tree"
 
-    """Represent a configuration space.
-    """
-
     def __init__(
             self,
             name: Union[str, None] = None,
             seed: Union[int, None] = None,
             meta: Optional[Dict] = None,
     ) -> None:
+        """
+        A configuration space organizes all hyperparameters and its conditions
```

Comment:
Could you please start with a one line description followed by a blank line and then a longer description?

Reply:
Is the following description okay:

> 

        A collection-like object containing a set of hyperparameter definitions and conditions.

        A configuration space organizes all hyperparameters and its conditions
        as well as its forbidden clauses. Configurations can be sampled from
        this configuration space. As underlying data structure, the
        configuration space uses a tree-based approach to represent the
        conditions and restrictions between hyperparameter.